src/Resources/app/storefront/src/checkout/checkout.plugin.js

```diff
@@ -104,40 +110,52 @@ export default class CheckoutPlugin extends Plugin {
         /* eslint-enable no-unused-vars */
     }
 
-    renderPaymentMethod (paymentMethod) {
-        //  if the container doesn't exist don't try to render the component
-        const paymentMethodContainer = $('[data-adyen-payment-method="' + adyenConfiguration.paymentMethodTypeHandlers[paymentMethod.type] + '"]');
+    onPaymentMethodChange(event) {
+        // Unmount any previously selected paymentMethod instance
+        if(!$.isEmptyObject(this.currentPaymentMethodInstance)) {
+            this.currentPaymentMethodInstance.unmount();
+            this.currentPaymentMethodInstance = {};
+        }
+        $('.adyen-payment-method-container-div').hide();
+        let paymentMethodContainer = $(`[data-adyen-payment-method-id="${$(event.target).val()}"]`);
 
-        if (paymentMethod.type in adyenConfiguration.componentsWithPayButton) {
-            // For payment methods with a direct pay button, the button is rendered on the confirm page
+        this.renderPaymentMethod(paymentMethodContainer);
+
+        paymentMethodContainer.show();
+    }
+
+    renderPaymentMethod(paymentMethodContainer) {
+        let handler = paymentMethodContainer.data('adyenPaymentMethod');
+        let type = Object.keys(adyenConfiguration.paymentMethodTypeHandlers)
+            .find(key => adyenConfiguration.paymentMethodTypeHandlers[key] === handler);
+
+        // Get the payment method object from paymentMethodsResponse
+        let filtered = $.grep(this.adyenCheckout.paymentMethodsResponse.paymentMethods, function(paymentMethod) {
+            return paymentMethod.type === type;
+        });
+        if (filtered.length === 0) {
             return;
         }
+        let paymentMethod = filtered[0];
 
-        // container doesn't exist, something went wrong on the template side
-        // If payment method doesn't have details, just skip it
-        if (!paymentMethodContainer || !paymentMethod.details) {
+        if (paymentMethod.type in adyenConfiguration.componentsWithPayButton) {
+            // For payment methods with a direct pay button, the button is rendered on the confirm page
             return;
         }
 
-        //Hide other payment method's contents when selecting an option
-        $('[name=paymentMethodId]').on("change", function () {
-            $('.adyen-payment-method-container-div').hide();
-            $(`[data-adyen-payment-method-id="${$(this).val()}"]`).show();
-        });
-
-        /*Use the storedPaymentMethod object and the custom onChange function as the configuration object together*/
+        // Mount payment method instance
         const configuration = Object.assign(paymentMethod, {
-            onChange: this.onPaymentMethodChange.bind(this)
+            onChange: this.onPaymentComponentChange.bind(this)
```

Comment:
Will this also show the "Update payment details" button when there's a state data already stored for a PM?

Reply:
Yes it will, that part remains unchanged and when the component is loaded it renders the preselected PM in the `init` method